client/sftp_client.py

```diff
@@ -196,202 +614,749 @@ def load_dirs(self, path, parent=None):
-
-	def signal_menu_activate_create_folder(self, _):
-		raise NotImplementedError('Parent create folder function not overriden')
+				self._tv_model.append(current, [None, None, None, None, None, None, None])
 
 	def signal_menu_toggled_hidden_files(self, _):  # pylint: disable=method-hidden
 		self.local_hidden = not self.local_hidden
 		self.refresh()
 
 	def signal_menu_activate_collapse_all(self, _):
-		self.treeview_local.collapse_all()
+		self.treeview.collapse_all()
 
 	def refresh(self):
-		model = self._tvmodel
+		model = self._tv_model
 		exp_lines = []
-		model.foreach(lambda model, path, iter: exp_lines.append(path) if self.treeview_local.row_expanded(path) else 0)
-		self.treeview_local.collapse_all()
+		model.foreach(lambda model, path, iter: exp_lines.append(path) if self.treeview.row_expanded(path) else 0)
+		self.treeview.collapse_all()
 		for path in exp_lines:
-			self.treeview_local.expand_row(path, False)
+			self.treeview.expand_row(path, False)
 
-	def signal_menu_activate_delete_prompt(self, _):
-		selection = self.treeview_local.get_selection()
+	def signal_menu_activate_create_folder(self, _):
+		selection = self.treeview.get_selection()
 		model, treeiter = selection.get_selected()
-		delete = gui_utilities.show_dialog_yes_no("Delete selected {0}?".format('directory' if model[treeiter][5] == -1 else 'file'), self.application.get_active_window())
-		if delete:
-			self.delete(model, treeiter)
+		if treeiter is not None:
+			fullname = model[treeiter][2]
+			path = self._tv_model.get_path(treeiter)
+		else:
+			fullname = self.default_directory
+		perm = self._check_perm(fullname)
+		w_perm = True if perm[4] == 'w' else False
+		if not w_perm:
+			gui_utilities.show_dialog_error('Permission Denied', self.application.get_active_window(), "Cannot access {0} as user".format(fullname))
+			return
+		if treeiter is None:
+			current = self._tv_model.append(treeiter, [' ', None, None, None, None, None, None])
+			self.rename(current)
+			return
+		if not self.treeview.row_expanded(path):
+			self.treeview.expand_row(path, False)
+		if self._tv_model.iter_children(treeiter) is None:
+			# If no children, one dummy node must be created to make row expandable and the other to be used as a placemark for new folder
+			self._tv_model.append(treeiter, [None, None, None, None, None, None, None])
+			current = self._tv_model.append(treeiter, [' ', None, None, None, None, None, None])
+			self.treeview.expand_row(path, False)
+		else:
+			current = self._tv_model.append(treeiter, [' ', None, None, None, None, None, None])
+		self.rename(current)
+
+	def signal_text_edited(self, renderer, path, text):
+		_iter = self._tv_model.get_iter(path)
+		parent = self._tv_model.iter_parent(_iter)
+		if parent is None:
+			new_path = self.default_directory + '/' + text
+		else:
+			new_path = self._tv_model[parent][2] + '/' + text
+		if text == ' ' or text == self._tv_model[_iter][0]:
```

Comment:
If you're going to ignore one space, you should ignore them all. Do `text = text.strip(); if not text...`